File: pallycon-drm-sdk/pallycon_drm_sdk_interface/lib/src/models/pallycon_content_configuration.dart
import '../enums/drm_type.dart';

class PallyConContentConfiguration {
  final String contentId;
  final String contentUrl;
  final DRMType? drmType;
  final String? token;
  final String? customData;
  final String? contentCookie;
  final Map<String, String>? contentHttpHeaders;
  final String? licenseCookie;
  final Map<String, String>? licenseHttpHeaders;
  final String licenseUrl;
  final String certificateUrl;
  final String? licenseCipherTablePath;

  PallyConContentConfiguration(this.contentId, this.contentUrl,
      {this.drmType,
      this.token,
      this.customData,
      this.contentCookie,
      this.contentHttpHeaders,
      this.licenseCookie,
      this.licenseHttpHeaders,
      this.licenseUrl =
          "https://license-global.pallycon.com/ri/licenseManager.do",
      this.certificateUrl = "",
      this.licenseCipherTablePath});
}

File: pallycon-drm-sdk/pallycon_drm_sdk_android/android/src/main/kotlin/com/pallycon/pallycon_drm_sdk_android/models/PallyConContentConfiguration.kt
package com.pallycon.pallycon_drm_sdk_android.models

import io.flutter.plugin.common.MethodCall

data class PallyConContentConfiguration(
    var contentUrl: String,
    var contentId: String,
    var token: String?,
    var customData: String?,
    var contentCookie: String?,
    var contentHttpHeaders: Map<String, String>?,
    var licenseCookie: String?,
    var licenseHttpHeaders: Map<String, String>?,
    var licenseUrl: String,
    var licenseCipherTablePath: String?
) {
    companion object {
        operator fun invoke(call: MethodCall): PallyConContentConfiguration? {
            val contentUrl = call.argument<String>("url")
            val contentId = call.argument<String>("contentId")
            val token = call.argument<String>("token")
            val customData = call.argument<String>("customData")
            val contentCookie = call.argument<String>("contentCookie")
            val contentHttpHeaders = call.argument<Map<String, String>>("contentHttpHeaders")
            val licenseCookie = call.argument<String>("licenseCookie")
            val licenseHttpHeaders = call.argument<Map<String, String>>("licenseHttpHeaders")
            val licenseUrl = call.argument<String>("licenseUrl")
            val licenseCipherTablePath = call.argument<String>("licenseCipherTablePath")
            if (contentUrl == null || contentId == null || licenseUrl == null || (token == null && customData == null)) {
                return null
            }

            return PallyConContentConfiguration(
                contentUrl,
                contentId,
                token,
                customData,
                contentCookie,
                contentHttpHeaders,
                licenseCookie,
                licenseHttpHeaders,
                licenseUrl,
                licenseCipherTablePath
            )
        }
    }
}